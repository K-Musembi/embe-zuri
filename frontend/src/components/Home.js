import React from "react";
import { useNavigate } from "react-router-dom";

const Home = () => {
    const navigate = useNavigate();

    const handleGetStarted = () => {
        navigate("/upload");
    }
    
    return (
        <div className="home-container">
            <h1>embe zuri</h1>
            <p>
                Welcome to embe zuri. This app enables early detection of mango fruit tree diseases, and offers solutions to mitigate them.
            </p>
            <button onClick={handleGetStarted}>Get Started</button>
        </div>
    );
};

export default Home;
