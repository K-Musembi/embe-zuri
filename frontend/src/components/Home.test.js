import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';

test('Home component snapshot test', () => {
    const tree = renderer.create(
        <MemoryRouter>
          <Home />
        </MemoryRouter>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
