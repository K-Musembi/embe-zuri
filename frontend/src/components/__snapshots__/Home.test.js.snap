// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot test 1`] = `
<div
  className="home-container"
>
  <h1>
    embe zuri
  </h1>
  <p>
    Welcome to embe zuri. This app enables early detection of mango fruit tree diseases, and offers solutions to mitigate them.
  </p>
  <button
    onClick={[Function]}
  >
    Get Started
  </button>
</div>
`;
